core/tx_pool.go

```diff
@@ -1272,6 +1282,7 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 	// Update all fork indicator by next pending block number.
 	next := new(big.Int).Add(newHead.Number, big.NewInt(1))
 	pool.istanbul = pool.chainconfig.IsIstanbul(next)
+	pool.donut = pool.chainconfig.IsDonut(next)
```

Comment:
do we call `validateTx` after this is enabled? Or old tx will continue to existt on the txpool?

Reply:
Good point!  `validateTx()` is only called when adding a transaction.  For later on it only checks a subset of what `validateTx()` checks, and this should be checked as well.

Reply:
I added a commit making it so that unprotected transactions are removed when Donut is activated.  I still need to add a test for it, though.